ignitionrobotics/ign-gazebo
# Model editor

# 🎉 New feature

## Summary

This pushes the model editor improvements into Fortress. The changes/improvements include:

1. Addition of  a "plus" button on the entity tree which supports model and light insertion.
2. Addition of an entity-aware "plus" button on the component inspector. This supports addition of links, joints, and light to a selected model. Sensors can be added to links.
3. Most sensor properties can be edited from the component inspector.
4. Editing of model, link, collision, visual poses is supported in the component inspector.
5. Edit boxes are simulation-state aware, and should be enabled when simulation is paused.
6. Changes are sent to the server when play or step is pressed.

## Test it

1. Open and empty world or a world like `shapes.sdf`
2. Use the tools to make edits. 
3. You can also open another gui instance `ign gazebo -v 4 -g`. Changes in this instance should be reflected when play/step is pressed in the first GUI instance.
4. Saving the world should also include your changes, assuming you have stepped or played simulation after making an edit.

## Checklist

**Note to maintainers**: Remember to use **Squash-Merge**
